In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns


In [21]:
df = pd.read_csv('../Data/analysis_ready.csv')
df.head()

,District,State,institutional_births,anc_4plus_visits,skilled_birth_attendance,postnatal_care_2days,female_literacy,child_marriage,health_insurance,clean_fuel,improved_sanitation,hh_electricity,csection_rate,women_anaemic,child_stunting,child_underweight,anc_first_trimester,institutional_births_public
0,Nicobars,Andaman & Nicobar Islands,97.81,71.65,98.60,85.09,87.54,11.36,2.65,56.93,83.45,97.90,11.52,38.30,21.59,24.60,62.77,96.65
1,North & Middle Andaman,Andaman & Nicobar Islands,97.70,79.19,98.33,92.53,84.01,15.38,2.07,61.25,86.38,93.24,12.91,62.07,27.02,42.80,74.46,95.01
2,South Andaman,Andaman & Nicobar Islands,99.46,85.94,96.86,88.13,86.67,17.06,1.15,91.92,89.33,99.58,37.14,57.66,21.11,17.43,79.40,83.83
3,Srikakulam,Andhra Pradesh,97.94,78.41,96.38,90.81,64.32,25.40,75.63,74.70,71.58,99.91,56.98,62.59,19.66,21.44,79.69,52.20
4,Vizianagaram,Andhra Pradesh,99.01,71.35,97.61,83.89,58.32,33.72,76.69,60.30,61.74,99.52,41.30,64.04,36.40,32.21,76.08,70.64


In [22]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 707 entries, 0 to 706
Data columns (total 18 columns):
 #   Column                       Non-Null Count  Dtype  
---  ------                       --------------  -----  
 0   District                     707 non-null    object 
 1   State                        707 non-null    object 
 2   institutional_births         707 non-null    float64
 3   anc_4plus_visits             707 non-null    float64
 4   skilled_birth_attendance     707 non-null    float64
 5   postnatal_care_2days         707 non-null    float64
 6   female_literacy              707 non-null    float64
 7   child_marriage               707 non-null    float64
 8   health_insurance             707 non-null    float64
 9   clean_fuel                   707 non-null    float64
 10  improved_sanitation          707 non-null    float64
 11  hh_electricity               707 non-null    float64
 12  csection_rate                707 non-null    float64
 13  women_anaemic       

In [23]:
df.describe()

,institutional_births,anc_4plus_visits,skilled_birth_attendance,postnatal_care_2days,female_literacy,child_marriage,health_insurance,clean_fuel,improved_sanitation,hh_electricity,csection_rate,women_anaemic,child_stunting,child_underweight,anc_first_trimester,institutional_births_public
count,707.000000,707.000000,707.000000,707.000000,707.000000,707.000000,707.000000,707.000000,707.000000,707.000000,707.000000,707.000000,707.000000,707.000000,707.000000,707.000000
mean,88.670382,60.325827,89.573494,78.703890,74.324696,20.462857,40.219208,54.086096,71.917496,96.996025,22.858444,55.943932,33.449448,29.432306,71.555870,64.940707
std,12.001523,20.768942,9.915865,15.958115,12.239847,12.522026,23.060809,24.178541,14.283175,4.354175,15.983400,11.966062,8.660313,9.973549,14.827496,16.315340
min,21.400000,-60.390000,30.940000,-97.480000,38.600000,-7.210000,1.150000,8.610000,29.240000,68.350000,1.380000,14.910000,-17.990000,-31.270000,-91.770000,18.110000
25%,83.685000,45.015000,84.875000,71.250000,66.860000,10.800000,20.285000,33.935000,62.000000,96.390000,10.680000,49.250000,27.290000,22.050000,62.465000,53.375000
50%,92.200000,62.470000,92.280000,83.060000,75.070000,18.270000,35.740000,50.420000,73.730000,98.650000,18.570000,57.180000,32.820000,29.260000,73.630000,66.530000
75%,97.395000,76.020000,96.945000,89.970000,83.705000,28.140000,59.940000,75.140000,83.310000,99.515000,30.865000,63.660000,39.175000,36.255000,82.160000,76.905000
max,100.000000,98.650000,100.000000,99.490000,99.720000,57.610000,97.790000,99.790000,99.850000,100.000000,82.400000,93.500000,60.610000,62.380000,99.600000,96.650000


In [24]:
df[df.institutional_births == 21.4]

,District,State,institutional_births,anc_4plus_visits,skilled_birth_attendance,postnatal_care_2days,female_literacy,child_marriage,health_insurance,clean_fuel,improved_sanitation,hh_electricity,csection_rate,women_anaemic,child_stunting,child_underweight,anc_first_trimester,institutional_births_public
413,Mon,Nagaland,21.4,9.66,30.94,25.62,78.22,2.98,39.26,13.0,86.44,96.36,1.38,30.23,35.51,23.34,44.58,20.15


In [25]:
df[df.institutional_births == 100]

,District,State,institutional_births,anc_4plus_visits,skilled_birth_attendance,postnatal_care_2days,female_literacy,child_marriage,health_insurance,clean_fuel,improved_sanitation,hh_electricity,csection_rate,women_anaemic,child_stunting,child_underweight,anc_first_trimester,institutional_births_public
139,South Goa,Goa,100.0,96.04,100.00,99.49,93.76,3.90,67.70,96.24,88.17,100.00,39.44,37.63,28.31,21.04,62.71,52.04
145,Porbandar,Gujarat,100.0,92.05,95.99,97.60,84.33,9.96,42.13,76.85,85.74,99.75,19.19,47.60,18.15,25.52,93.54,61.15
267,Tumkur,Karnataka,100.0,80.35,93.07,95.21,81.87,24.80,30.95,85.26,86.14,99.60,52.11,49.28,40.31,26.72,68.58,66.86
270,Hassan,Karnataka,100.0,75.77,97.10,92.61,82.04,16.15,36.89,86.13,87.87,99.28,41.80,42.22,27.12,23.81,67.63,71.70
271,Dakshina Kannada,Karnataka,100.0,82.00,91.89,83.50,92.72,4.89,36.29,82.04,97.10,99.83,30.40,38.75,25.10,26.35,86.14,41.85
273,Mysore,Karnataka,100.0,85.74,97.82,93.54,78.92,17.50,26.24,94.73,86.44,99.47,43.65,47.97,27.50,28.23,82.02,56.17
274,Chamarajanagar,Karnataka,100.0,84.11,96.70,94.16,72.36,19.29,34.37,88.96,75.16,98.65,31.41,46.32,32.21,28.73,83.18,82.65
279,Bangalore Rural,Karnataka,100.0,90.92,97.92,91.68,83.81,14.11,34.88,93.56,88.80,99.22,43.58,44.24,36.57,23.84,81.82,72.04
280,Ramanagara,Karnataka,100.0,88.74,100.00,86.95,82.67,11.81,35.71,91.98,89.63,99.61,39.91,45.46,15.62,19.80,91.84,68.78
281,Kasaragod,Kerala,100.0,91.18,100.00,95.03,95.91,4.74,44.01,73.50,98.85,99.46,28.75,35.35,25.34,21.42,97.42,29.86


In [26]:
df[['institutional_births', 'female_literacy', 'child_marriage', 'health_insurance', 'improved_sanitation']].corr()

,institutional_births,female_literacy,child_marriage,health_insurance,improved_sanitation
institutional_births,1.000000,0.279512,-0.185882,0.181116,0.188542
female_literacy,0.279512,1.000000,-0.604752,0.013354,0.665533
child_marriage,-0.185882,-0.604752,1.000000,0.012638,-0.567261
health_insurance,0.181116,0.013354,0.012638,1.000000,0.077843
improved_sanitation,0.188542,0.665533,-0.567261,0.077843,1.000000


In [27]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 707 entries, 0 to 706
Data columns (total 18 columns):
 #   Column                       Non-Null Count  Dtype  
---  ------                       --------------  -----  
 0   District                     707 non-null    object 
 1   State                        707 non-null    object 
 2   institutional_births         707 non-null    float64
 3   anc_4plus_visits             707 non-null    float64
 4   skilled_birth_attendance     707 non-null    float64
 5   postnatal_care_2days         707 non-null    float64
 6   female_literacy              707 non-null    float64
 7   child_marriage               707 non-null    float64
 8   health_insurance             707 non-null    float64
 9   clean_fuel                   707 non-null    float64
 10  improved_sanitation          707 non-null    float64
 11  hh_electricity               707 non-null    float64
 12  csection_rate                707 non-null    float64
 13  women_anaemic       

In [28]:
df.isnull().sum()

District                       0
State                          0
institutional_births           0
anc_4plus_visits               0
skilled_birth_attendance       0
postnatal_care_2days           0
female_literacy                0
child_marriage                 0
health_insurance               0
clean_fuel                     0
improved_sanitation            0
hh_electricity                 0
csection_rate                  0
women_anaemic                  0
child_stunting                 0
child_underweight              0
anc_first_trimester            0
institutional_births_public    0
dtype: int64

In [29]:
df.dropna(inplace=True)

In [30]:
df.isnull().sum()

District                       0
State                          0
institutional_births           0
anc_4plus_visits               0
skilled_birth_attendance       0
postnatal_care_2days           0
female_literacy                0
child_marriage                 0
health_insurance               0
clean_fuel                     0
improved_sanitation            0
hh_electricity                 0
csection_rate                  0
women_anaemic                  0
child_stunting                 0
child_underweight              0
anc_first_trimester            0
institutional_births_public    0
dtype: int64

In [31]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 707 entries, 0 to 706
Data columns (total 18 columns):
 #   Column                       Non-Null Count  Dtype  
---  ------                       --------------  -----  
 0   District                     707 non-null    object 
 1   State                        707 non-null    object 
 2   institutional_births         707 non-null    float64
 3   anc_4plus_visits             707 non-null    float64
 4   skilled_birth_attendance     707 non-null    float64
 5   postnatal_care_2days         707 non-null    float64
 6   female_literacy              707 non-null    float64
 7   child_marriage               707 non-null    float64
 8   health_insurance             707 non-null    float64
 9   clean_fuel                   707 non-null    float64
 10  improved_sanitation          707 non-null    float64
 11  hh_electricity               707 non-null    float64
 12  csection_rate                707 non-null    float64
 13  women_anaemic       

In [32]:
df.head()

,District,State,institutional_births,anc_4plus_visits,skilled_birth_attendance,postnatal_care_2days,female_literacy,child_marriage,health_insurance,clean_fuel,improved_sanitation,hh_electricity,csection_rate,women_anaemic,child_stunting,child_underweight,anc_first_trimester,institutional_births_public
0,Nicobars,Andaman & Nicobar Islands,97.81,71.65,98.60,85.09,87.54,11.36,2.65,56.93,83.45,97.90,11.52,38.30,21.59,24.60,62.77,96.65
1,North & Middle Andaman,Andaman & Nicobar Islands,97.70,79.19,98.33,92.53,84.01,15.38,2.07,61.25,86.38,93.24,12.91,62.07,27.02,42.80,74.46,95.01
2,South Andaman,Andaman & Nicobar Islands,99.46,85.94,96.86,88.13,86.67,17.06,1.15,91.92,89.33,99.58,37.14,57.66,21.11,17.43,79.40,83.83
3,Srikakulam,Andhra Pradesh,97.94,78.41,96.38,90.81,64.32,25.40,75.63,74.70,71.58,99.91,56.98,62.59,19.66,21.44,79.69,52.20
4,Vizianagaram,Andhra Pradesh,99.01,71.35,97.61,83.89,58.32,33.72,76.69,60.30,61.74,99.52,41.30,64.04,36.40,32.21,76.08,70.64


In [33]:
import pandas as pd
import statsmodels.api as sm
import statsmodels.formula.api as smf
from statsmodels.stats.diagnostic import het_breuschpagan
from statsmodels.stats.outliers_influence import variance_inflation_factor

# Load pivoted data

df = df.dropna(subset=['institutional_births', 'female_literacy', 'child_marriage',
                        'health_insurance', 'improved_sanitation', 'clean_fuel'])

print(f"Districts with complete data: {len(df)}")
print(df[['institutional_births', 'female_literacy', 'child_marriage',
           'health_insurance', 'improved_sanitation']].describe().round(2))

# ============================================================
# MODEL 1: Simple bivariate — literacy alone
# ============================================================
m1 = smf.ols('institutional_births ~ female_literacy', data=df).fit(cov_type='HC1')
print("\n=== MODEL 1: Bivariate ===")
print(m1.summary2().tables[1].round(4))
print(f"R-squared: {m1.rsquared:.4f}")

# ============================================================
# MODEL 2: Multiple controls, no state FE
# ============================================================
m2 = smf.ols('''institutional_births ~ female_literacy + child_marriage 
                + health_insurance + improved_sanitation + clean_fuel''', data=df).fit(cov_type='HC1')
print("\n=== MODEL 2: Multiple Controls ===")
print(m2.summary2().tables[1].round(4))
print(f"R-squared: {m2.rsquared:.4f}")

# ============================================================
# MODEL 3: State Fixed Effects
# ============================================================
m3 = smf.ols('''institutional_births ~ female_literacy + child_marriage 
                + health_insurance + improved_sanitation + clean_fuel
                + C(State)''', data=df).fit(cov_type='HC1')
print("\n=== MODEL 3: State Fixed Effects ===")
# Print only non-state-dummy coefficients
coefs = m3.summary2().tables[1]
non_state = coefs[~coefs.index.str.startswith('C(State)')]
print(non_state.round(4))
print(f"R-squared: {m3.rsquared:.4f}")

# ============================================================
# DIAGNOSTICS
# ============================================================
print("\n=== DIAGNOSTICS (Model 2) ===")

# VIF
X = df[['female_literacy', 'child_marriage', 'health_insurance', 
        'improved_sanitation', 'clean_fuel']].dropna()
X_const = sm.add_constant(X)
vif = pd.DataFrame({
    'Variable': X.columns,
    'VIF': [variance_inflation_factor(X_const.values, i+1) for i in range(len(X.columns))]
})
print("\nVIF (>10 = problem):")
print(vif.to_string(index=False))

# Breusch-Pagan heteroskedasticity test
bp_test = het_breuschpagan(m2.resid, m2.model.exog)
print(f"\nBreusch-Pagan p-value: {bp_test[1]:.4f}")
print(f"  {'Heteroskedasticity detected → robust SEs already applied' if bp_test[1] < 0.05 else 'Homoskedastic'}")

# RESET test (Ramsey)
from statsmodels.stats.diagnostic import linear_reset
reset = linear_reset(m2, power=3, use_f=True)
print(f"\nRESET test p-value: {reset.pvalue:.4f}")
print(f"  {'Possible specification error' if reset.pvalue < 0.05 else 'No specification error detected'}")

# ============================================================
# COMPARISON TABLE
# ============================================================
print("\n=== MODEL COMPARISON ===")
print(f"{'':30s} {'M1 (Bivariate)':>15s} {'M2 (Controls)':>15s} {'M3 (State FE)':>15s}")
print("-" * 78)
for var in ['female_literacy', 'child_marriage', 'health_insurance', 'improved_sanitation', 'clean_fuel']:
    vals = []
    for m in [m1, m2, m3]:
        if var in m.params:
            coef = m.params[var]
            pval = m.pvalues[var]
            stars = '***' if pval < 0.01 else '**' if pval < 0.05 else '*' if pval < 0.1 else ''
            vals.append(f"{coef:.3f}{stars}")
        else:
            vals.append("—")
    print(f"{var:30s} {vals[0]:>15s} {vals[1]:>15s} {vals[2]:>15s}")
print("-" * 78)
print(f"{'R-squared':30s} {m1.rsquared:>15.4f} {m2.rsquared:>15.4f} {m3.rsquared:>15.4f}")
print(f"{'N':30s} {int(m1.nobs):>15d} {int(m2.nobs):>15d} {int(m3.nobs):>15d}")

Districts with complete data: 707
       institutional_births  female_literacy  child_marriage  \
count                707.00           707.00          707.00   
mean                  88.67            74.32           20.46   
std                   12.00            12.24           12.52   
min                   21.40            38.60           -7.21   
25%                   83.68            66.86           10.80   
50%                   92.20            75.07           18.27   
75%                   97.40            83.70           28.14   
max                  100.00            99.72           57.61   

       health_insurance  improved_sanitation  
count            707.00               707.00  
mean              40.22                71.92  
std               23.06                14.28  
min                1.15                29.24  
25%               20.28                62.00  
50%               35.74                73.73  
75%               59.94                83.31  
max          

c:\Users\Admin\anaconda3\Lib\site-packages\statsmodels\base\model.py:1894: ValueWarning: covariance of constraints does not have full rank. The number of constraints is 40, but rank is 39
  warnings.warn('covariance of constraints does not have full '


# Regression Results: Determinants of Institutional Delivery Across 707 Indian Districts

## Key Findings

### 1. Female Literacy is the Strongest and Most Robust Predictor

Female literacy is statistically significant at the 1% level across all three model specifications, with coefficients of 0.274 (bivariate), 0.145 (controls), and 0.330 (state FE). In the full state fixed-effects model, a 10 percentage-point increase in female literacy is associated with a 3.3 percentage-point increase in institutional delivery rates, holding all else constant. The coefficient dips in Model 2 (0.145) when correlated controls like clean fuel absorb shared variance, but rebounds to 0.330 once state fixed effects partial out between-state heterogeneity — revealing literacy's true within-state explanatory power.

**Policy implication:** Investments in girls' education yield downstream returns in maternal healthcare utilization. Districts with the lowest female literacy rates should be co-targeted for both education and maternal health interventions.

### 2. Clean Fuel Usage as a Proxy for Household Modernization

Clean fuel for cooking (LPG, electricity) emerges as the second-strongest predictor (coefficient: 0.161, p<0.01 in Model 3). This variable proxies for household wealth, connectivity to government distribution networks (e.g., Ujjwala Yojana), and broader modernization — all of which correlate with healthcare-seeking behavior. The coefficient attenuates from 0.277 (Model 2) to 0.161 (Model 3) upon adding state fixed effects, indicating that a substantial portion of the clean fuel–delivery association operates through state-level development differences.

**Policy implication:** Infrastructure development and welfare scheme penetration indirectly improve maternal health outcomes. Cross-sectoral convergence between energy access programs and maternal health programs could amplify impact.

### 3. Health Insurance Emerges as a Significant Predictor

Health insurance coverage is statistically significant in both Model 2 (0.121, p<0.01) and Model 3 (0.137, p<0.01). Unlike clean fuel, the coefficient actually increases after adding state fixed effects, suggesting that within-state variation in insurance coverage independently predicts institutional delivery. A 10 percentage-point increase in health insurance coverage is associated with a 1.4 percentage-point increase in institutional delivery rates. This likely captures the effect of schemes like Ayushman Bharat and state-specific health insurance programs that reduce financial barriers to facility-based delivery.

**Policy implication:** Expanding health insurance coverage — particularly in districts with low enrollment — can directly improve institutional delivery uptake by removing out-of-pocket cost barriers.

### 4. Child Marriage Effect is Statistically Null

Child marriage is statistically insignificant in both Model 2 (-0.008, p=0.83) and Model 3 (0.002, p=0.96). With the full 707-district dataset, the variable shows no independent association with institutional delivery in any specification. Child marriage and institutional delivery are correlated at the state level, but district-level child marriage rates have no independent predictive power.

**Policy implication:** Addressing child marriage requires state-level policy interventions (enforcement of PCMA, awareness campaigns) rather than district-level targeting within maternal health programs.

### 5. Sanitation's Negative Sign Attenuates and Loses Significance

Improved sanitation carries a negative coefficient (-0.057, p=0.10 in Model 3), no longer statistically significant at conventional levels. In the previous 340-district analysis (Phase 1 states only), this was significant (p<0.05) and posed an interpretation challenge. With the full dataset, the effect washes out — suggesting the earlier result was driven by sample composition. The residual negative direction may reflect ceiling effects in urbanized districts, but the lack of significance means no strong inference should be drawn.

### 6. State-Level Policy Variation Explains the Majority of Disparities

R-squared increases from 0.34 (Model 2) to 0.75 (Model 3) upon adding state fixed effects. This 41 percentage-point jump indicates that unobserved state-level factors are the dominant driver of institutional delivery variation — including differential implementation of Janani Suraksha Yojana (JSY), National Health Mission (NHM) funding allocations, state health budgets, ASHA worker density, and political prioritization of maternal health.

**Policy implication:** The state a woman resides in is the single largest determinant of whether she delivers in an institution. Reducing interstate disparities in maternal health program implementation should be a central focus of national policy. States like Nagaland, Meghalaya, and Arunachal Pradesh require disproportionate attention and resource allocation.

## Model Comparison Table

| Variable | M1 (Bivariate) | M2 (Controls) | M3 (State FE) |
|----------|:--------------:|:--------------:|:--------------:|
| Female Literacy | 0.274*** | 0.145*** | 0.330*** |
| Child Marriage | — | -0.008 | 0.002 |
| Health Insurance | — | 0.121*** | 0.137*** |
| Improved Sanitation | — | -0.160*** | -0.057 |
| Clean Fuel | — | 0.277*** | 0.161*** |
| R-squared | 0.078 | 0.341 | 0.754 |
| N | 707 | 707 | 707 |

Note: *** p<0.01. Robust (HC1) standard errors used throughout.

## Diagnostics Summary

| Test | Result | Interpretation |
|------|--------|----------------|
| VIF (all variables) | All below 2.2 | No multicollinearity concerns |
| Breusch-Pagan | p < 0.001 | Heteroskedasticity present; robust (HC1) standard errors applied |
| RESET (Ramsey) | p < 0.001 | Possible specification error; consider adding quadratic terms or interaction effects in future iterations |

## Limitations

This analysis uses district-level aggregate data from NFHS-5 factsheets covering all 707 districts across 36 states and union territories. Results represent ecological associations, not individual-level causal effects. The RESET test indicates potential specification issues, suggesting that the relationship between predictors and institutional delivery may be non-linear or that relevant interaction terms are omitted. The constraints warning in Model 3 reflects automatic dropping of one state as the reference category — standard fixed-effects behavior that does not affect coefficient estimates. Individual-level analysis using DHS microdata (pending access approval) will address ecological inference limitations through logistic regression with household-level controls and survey weights.